de1plus/dui.tcl

```diff
@@ -5441,52 +5455,80 @@ namespace eval ::dui {
+		        {slider_change {}} args } {
+            variable sliders
+            if { ![info exists sliders(${page},${dscale_tag})] } {
+                msg -WARNING [namespace current] scale_moveto: "scale on page '$page' with tag '$dscale_tag' not found"
+                return
+            }
+            set offset $sliders(${page},${dscale_tag})
+            # If we're in the middle of a slider drag motion and are here due to the trace add variable, do nothing.
+            if { $slider_coord eq "" && $offset ne "" } return
+            if { $offset eq "" } {
+                set offset 0
+            }
+            
+            set can [dui canvas]
 			set slider [dui item get $page "${dscale_tag}-crc"]
 			if { $slider eq "" } return
 			lassign [$can bbox $slider] sx0 sy0 sx1 sy1
 			# Return if the page is not currently visible
-			if {$sx0 eq "" } return		
+			if {$sx0 eq "" } return	
 			
 			set back [dui item get $page "${dscale_tag}-bck"]
 			set front [dui item get $page "${dscale_tag}-frn"]
-			if { $slider eq "" || $back eq "" || $front eq "" } {
+			if { $back eq "" || $front eq "" } {
 				return
 			}
-			
+
 			lassign [$can coords $back] bx0 by0 bx1 by1
 			lassign [$can coords $front] fx0 fy0 fx1 fy1
 				
 			set swidth [expr {$sx1-$sx0}]
 			set sheight [expr {$sy1-$sy0}]
 			if { ($bx1 - $bx0) >= ($by1 -$by0) } {
 				set orient h
+                if { $slider_coord ne "" } {
+                    set slider_coord [dui::platform::translate_coordinates_finger_down_x $slider_coord]
+                }
 			} else {
 				set orient v
+                if { $slider_coord ne "" } {
+                    set slider_coord [dui::platform::translate_coordinates_finger_down_y $slider_coord]
+                }
 			}
-			
+                    
 			set varvalue ""
 			# If no slider_coord is given, reads the value from whatever is in variable $varname and transforms it
 			#	to a slider_coord value.
 			if { $slider_coord eq "" && $varname ne "" } {
 				if { [info exists $varname] } {
-					set varvalue [number_in_range [subst \$$varname] 0 $from $to $resolution $n_decimals]  
+					set varvalue [number_in_range [subst \$$varname] 0 $from $to $resolution $n_decimals]
 				} else {
 					set varvalue [number_in_range $from 0 $from $to $resolution $n_decimals]
 					set $varname $varvalue
 				}
-
+                
 				if { [string is double -strict $varvalue] } {
-					if { $varvalue <= $from } {
-						switch $orient h {set slider_coord $bx0} v {set slider_coord [expr {$by1-$sheight/2}]}
-					} elseif { $varvalue >= $to } {
-						switch $orient h {set slider_coord [expr {$bx1-$swidth/2}]} v {set slider_coord $by0 }
-					} elseif { $orient eq "h" } {
-						set slider_coord [expr {($bx0+$swidth/2)+($bx1-$bx0-$swidth)*$varvalue/($to-$from)}]
-					} else {
-						set slider_coord [expr {($by1-$sheight/2)+($by1-$by0-$sheight)*$varvalue/($from-$to)}]
-					}
+#					if { $varvalue <= $from } {

```

Comment:
Can we remove that code to (hopefully) some day have less blocks of commented code?

Reply:
Please let me leave this (and the others below) there for the initial testing. I will remove it as soon as this is a bit more tested and I'm sure we want it like this. I will remove all comments before the code promotes to stable. I already did a cleaning commit a few weeks ago to remove most commented code in DUI, and I will do a general cleanup before stable, promised.